autogen.pl

```diff
@@ -1096,6 +1098,7 @@ sub in_tarball {
   --no-ompi | -no-ompi          Do not build the Open MPI layer
   --no-orte | -no-orte          Do not build the ORTE layer
   --no-oshmem | -no-oshmem      Do not build the OSHMEM layer
+  --libmpi-name | -libmpi-name  Override the default lib name of: lib$libmpi_name_arg
```

Comment:
I guess my default position on this is: we only put stuff in `autogen.pl` that *can't* go in `configure.ac`.  Otherwise, you could make an argument that just about any `configure` CLI option should also (or only) be available in `autogen.pl`.

Indeed, we already have some branding-specific `configure` CLI options.  For example, how is the library name different than `--with-package-string=STRING` or `--with-ident-string=STRING`?